DHL/Datatype/Base.php

```diff
@@ -222,10 +222,11 @@ public function __call($name, $arguments)
             $key .= 's';
         }
 
-        if (!array_key_exists($key, $this->_values))
-        {
-            throw new \InvalidArgumentException('Field : ' . $key . ' is not defined for ' . get_class($this));
-        }
+        /** M Ward - removed this 5/3/19 */
```

Comment:
First: Commented out code should never be committed into a repo (except in special circumstances). When not using a repo, devs often simply comment out code to serve as a record as what they did, and so they can "revert" their change by uncommenting it. This is unnecessary inside a repo, because deleted code can always be viewed and resurrected from the commit log.

Second: I don't like that this was commented out/deleted/removed. Here it is enforcing that the value you are trying to set is supported by the API class. For example, look at the [Piece datatype](https://github.com/optimalship/DHL-API/blob/master/DHL/Datatype/GB/Piece.php). You can see how it declares its supported API values:

```php
    protected $_params = array(
        'PieceID' => array(
            'type' => 'PieceID',
            'required' => false,
            'subobject' => false,
            'comment' => 'Piece ID',
            'maxLength' => '35',
        ), 
        'PackageType' => array(
// ...
```
That class, `Piece`, extends this `Base` class, such that when you call `(new Piece())->PackageType = 'foo'`, it is invoking this magic PHP method as `__call('PackageType', 'foo')`.

The deleted logic here would ensure that if you call `(new Piece())->NonExistentKey = 'bar'`, you will get an `\InvalidArgumentException`. This is good logic; it enforces that you are supplying it only known values.

What I suspect fueled this change is DHL added some new field to some API object, such as `Piece.NewField`, that the original repo author had not added to his PHP class. As such, when you guys tried to supply this as `(new Piece())->NewField`, you got an error. Commenting out this code allowed this to function; however, it was at the expense of type safety and data integrity.

The proper way to have supported this `NewField` would have been to edit the appropriate PHP class and add it to the `$_params` class property. That is not as easy, but it is more robust and preserves this now-deleted validation logic.

*Conclusion*: Uncomment this code to restore it to the original code. If we plug this forked library into the existing project and experience an error, we will properly add the new field to the appropriate PHP class, commit, push, merge, and update.